dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+          make_string("The number of elements passed as `", offsetArgName, "` and `", sizeArgName,
+                      "` must match for corresponding samples. However for sample of idx ",
+                      sample_idx, " there are ", num_chunks, " offsets and ",
+                      sizes.shape[sample_idx].num_elements(), " sizes."));
+    }
+    for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+      auto num_chunks = offsets.shape[sample_idx].num_elements();
+      auto sample_num_bytes = in_shape[sample_idx].num_elements();
+      const auto *chunk_offsets = offsets.data[sample_idx];
+      const auto *chunk_sizes = sizes.data[sample_idx];
+      for (int chunk_idx = 0; chunk_idx < num_chunks; chunk_idx++) {
+        int64_t chunk_end = chunk_offsets[chunk_idx];
+        chunk_end += chunk_sizes[chunk_idx];
+        DALI_ENFORCE(
+            chunk_end <= sample_num_bytes,
+            make_string("Input chunk cannot exceed the sample size. However, for a sample of idx ",
+                        sample_idx, ", got a chunk that starts at the position ",
+                        chunk_offsets[chunk_idx], " and has size ", chunk_sizes[chunk_idx],
+                        " while the sample contains only ", sample_num_bytes, " bytes."));
+      }
+    }
+  }
+
+  template <typename TLShape>
+  void InferSizesFromOffsets(const TLShape &in_shape) {
+    assert(offsets_.size() == chunks_per_sample_.num_elements());
+    auto last_chunk_error_msg = [](auto sample_idx) {
+      return make_string(
+          "The last chunk's offset exceeds the size of the sample for sample of idx ", sample_idx,
+          ".");
+    };
+    auto non_monotone_error = [](auto sample_idx) {
+      return make_string("If the `", offsetArgName, "` argument is specified and the `",
+                         sizeArgName, "` is not, the offsets must be strictly increasing. ",
+                         "The inferred size of a chunk would be non-positive for sample of idx ",
+                         sample_idx, ".");
+    };
+    auto batch_size = in_shape.num_samples();
+    sizes_.clear();
+    sizes_.reserve(chunks_per_sample_.num_elements());
+    for (int64_t sample_idx = 0, flat_chunk_idx = 0; sample_idx < batch_size; sample_idx++) {
+      auto num_chunks = chunks_per_sample_[sample_idx].num_elements();
+      if (num_chunks == 0) {
+        continue;
+      }
+      auto in_chunk_size = in_shape[sample_idx].num_elements();
+      for (int chunk_idx = 0; chunk_idx < num_chunks; chunk_idx++, flat_chunk_idx++) {
+        bool is_last_chunk = chunk_idx == num_chunks - 1;
+        int64_t next_chunk_offset = is_last_chunk ? in_chunk_size : offsets_[flat_chunk_idx + 1];
+        int64_t chunk_size = next_chunk_offset - offsets_[flat_chunk_idx];
+        DALI_ENFORCE(chunk_size > 0, is_last_chunk ? last_chunk_error_msg(sample_idx) :
+                                                     non_monotone_error(sample_idx));
```

Comment:
This cannot happen for the last chunk - offset validation takes care of that.
```suggestion
        assert(chunk_size > 0 || !is_last_chunk);
        DALI_ENFORCE(chunk_size > 0, non_monotone_error(sample_idx));
```

Reply:
Nice catch.